File: src/test/java/kitchenpos/menus/tobe/domain/MenuFixtures.java
package kitchenpos.menus.tobe.domain;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;
import kitchenpos.menus.tobe.domain.vo.MenuGroupName;
import kitchenpos.menus.tobe.domain.vo.MenuName;
import kitchenpos.menus.tobe.domain.vo.MenuPrice;
import kitchenpos.menus.tobe.domain.vo.MenuProductQuantity;
import kitchenpos.products.application.FakePurgomalumClient;

public final class MenuFixtures {

    private MenuFixtures() {
    }

    public static MenuGroup menuGroup(String name) {
        return new MenuGroup(new MenuGroupName(name));
    }

    public static MenuName menuName(String name) {
        return new MenuName(name, new FakePurgomalumClient());
    }

    public static MenuPrice menuPrice(BigDecimal price, MenuProducts menuProducts) {
        return new MenuPrice(price, menuProducts);
    }

    public static MenuProduct menuProduct(long quantity, BigDecimal productPrice) {
        return new MenuProduct(
                UUID.randomUUID(),
                new MenuProductQuantity(quantity),
                productPrice
        );
    }

    public static MenuProducts menuProducts(MenuProduct... menuProducts) {
        return new MenuProducts(List.of(menuProducts));
    }
}

File: src/test/java/kitchenpos/fixture/MenuFixtures.kt
package kitchenpos.fixture

import kitchenpos.common.Price
import kitchenpos.common.price
import kitchenpos.menus.application.tobe.FakeMenuNameValidator
import kitchenpos.menus.tobe.domain.*
import kitchenpos.menus.tobe.dto.`in`.MenuCreateRequest
import kitchenpos.menus.tobe.dto.`in`.MenuProductCreateRequest
import java.math.BigDecimal
import java.util.*

object MenuFixtures {
    fun menu(
        price: Price = BigDecimal.valueOf(15000).price(),
        menuProducts: MenuProducts = MenuProducts(listOf(menuProduct())),
        menuNameValidator: MenuNameValidator = FakeMenuNameValidator,
        displayStatus: Boolean = true
    ): Menu {
        return Menu.of(
            menuGroup = menuGroup(),
            name = MenuName.of("양념치킨", menuNameValidator),
            price = price,
            displayStatus = displayStatus,
            menuProducts = menuProducts,
        )
    }

    fun menuCreateRequest(
        menuGroupId: UUID,
        name: String = "양념치킨",
        price: Price = BigDecimal.valueOf(1000).price(),
        displayStatus: Boolean = true,
        menuProducts: List<MenuProductCreateRequest>
    ): MenuCreateRequest {
        return MenuCreateRequest(
            name = name,
            price = price,
            menuGroupId = menuGroupId,
            displayed = displayStatus,
            menuProducts = menuProducts
        )
    }

    fun menuGroup(): MenuGroup {
        return MenuGroup(MenuGroupName("치킨"))
    }

    fun menuProduct(
        productId: UUID = ProductFixtures.product().id,
        quantity: MenuProductQuantity = MenuProductQuantity(1),
        price: Price = BigDecimal.valueOf(20000).price()
    ): MenuProduct {
        return MenuProduct(productId = productId, quantity = quantity, price = price)
    }
}
